### ISSUE TITLE
Null pointer dereference vulnerability in ifilter_bank(libfaad/filtbank.c:307) 

### ISSUE THREAD

**Issue by fantasy7082** [knik0/faad2#27]
Hi, i found a null pointer dereference bug in Freeware Advanced Audio Decoder 2 (FAAD2) 2.8.8. It crashed in function ifilter_bank.the details are below(ASAN):

```
./faad faad_res/011-null-point-filtbank_307 -o out.wav
 *********** Ahead Software MPEG-4 AAC Decoder V2.8.8 ******************

 Build: Dec 13 2018
 Copyright 2002-2004: Ahead Software AG
 http://www.audiocoding.com
 bug tracking: https://sourceforge.net/p/faac/bugs/
 Floating point version

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License.

 **************************************************************************

faad_res/011-null-point-filtbank_307 file info:
ADTS, 0.043 sec, 37 kbps, 48000 Hz

  ---------------------
 | Config:  2 Ch       |
  ---------------------
 | Ch |    Position    |
  ---------------------
 | 00 | Left front     |
 | 01 | Right front    |
  ---------------------

ASAN:SIGSEGVfaad_res/011-null-point-filtbank_307.
=================================================================
==7079==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f94bfdbee3d bp 0x7fff8addf2e0 sp 0x7fff8addd1f0 T0)
    #0 0x7f94bfdbee3c in ifilter_bank /root/faad2_asan/libfaad/filtbank.c:307
    #1 0x7f94bfdf819d in reconstruct_channel_pair /root/faad2_asan/libfaad/specrec.c:1258
    #2 0x7f94bfdfe823 in channel_pair_element /root/faad2_asan/libfaad/syntax.c:759
    #3 0x7f94bfdfccbf in decode_cpe /root/faad2_asan/libfaad/syntax.c:402
    #4 0x7f94bfdfd398 in raw_data_block /root/faad2_asan/libfaad/syntax.c:448
    #5 0x7f94bfdb79c3 in aac_frame_decode /root/faad2_asan/libfaad/decoder.c:990
    #6 0x7f94bfdb7566 in NeAACDecDecode /root/faad2_asan/libfaad/decoder.c:821
    #7 0x40f8ae in decodeAACfile /root/faad2_asan/frontend/main.c:679
    #8 0x411dd4 in faad_main /root/faad2_asan/frontend/main.c:1323
    #9 0x411fe5 in main /root/faad2_asan/frontend/main.c:1366
    #10 0x7f94bf9ef82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #11 0x401aa8 in _start (/usr/local/faad-asan/bin/faad+0x401aa8)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /root/faad2_asan/libfaad/filtbank.c:307 ifilter_bank
==7079==ABORTING
```

POC FILE:https://github.com/fantasy7082/image_test/blob/master/011-null-point-filtbank_307

**Comment by hlef** [comment]
Same underlying issue as #26.

@fabiangreffrath : fixed by 466b01d504d7e45f1e9169ac90b3e34ab94aed14.

**Comment by fabiangreffrath** [comment]
Closing, thanks!
